src/main/java/net/neoforged/neoforge/transfer/handlers/resources/IResourceHandler.java

```diff
@@ -0,0 +1,230 @@
+     *
+     * @param index The index to get the resource from.
+     */
+    T getResource(int index);
+
+    /**
+     * Returns the amount of the {@link #getResource currently stored resource} at the given index.
+     *
+     * <p>In general, resource handlers can report {@code long} amounts.
+     * However, if the handler is known to only support amounts up to {@code Integer.MAX_VALUE},
+     * or if the caller prefers to deal in {@code int}s only,
+     * the {@link #getAmountAsInt int-returning overload} can be used instead.
+     *
+     * <p>The returned amount should be <strong>non-negative</strong>, and should never surpass the {@link #getCapacity capacity} of the same index.
+     *
+     * @param index The index to get the amount from.
+     * @return the amount at the given index
+     * @see #getAmountAsInt(int)
+     */
+    long getAmount(int index);
+
+    /**
+     * Returns the amount of the {@link #getResource currently stored resource} at the given index, as an {@code int}.
+     *
+     * <p>This is a convenience method to clamp the amount to an {@code int},
+     * for the cases where the handler is known to only support amounts up to {@code Integer.MAX_VALUE},
+     * or if the caller prefers to deal in {@code int}s only.
+     *
+     * <p>The returned amount should be <strong>non-negative</strong>, and should never surpass the {@link #getCapacityAsInt capacity} of the same index.
+     *
+     * @param index The index to get the amount from.
+     * @return the amount at the given index, as an int
+     * @see #getAmount(int) the long-returning overload
+     */
+    @ApiStatus.NonExtendable
+    default int getAmountAsInt(int index) {
+        return Ints.saturatedCast(getAmount(index));
+    }
+
+    /**
+     * Returns the capacity of the handler at the given index and for the given resource,
+     * irrespective of the current amount or resource currently at that index.
+     * <p>
+     * In general, resource handlers can report {@code long} capacities.
+     * However, if the handler is known to only support capacities up to {@code Integer.MAX_VALUE},
+     * or if the caller prefers to deal in {@code int}s only,
+     * the {@link #getCapacityAsInt int-returning overload} can be used instead.
+     * <p>
+     * While passing in resources that would return {@code false} on {@link #isValid(int, IResource)}, it should be expected to always return 0.
+     * <p>
+     * This function serves as metadata only, and its result might be approximate. The only way to know if a handler will accept a resource, is to try to {@link #insert insert} it.
+     *
+     * @param index    The index to get the limit for.
+     * @param resource The resource to get the limit for. May be empty to return a generic limit.
+     * @return the capacity at the given index
+     * @see #getCapacityAsInt(int, IResource)
+     */
+    // TODO: remark that the amount should be larger than the capacity?
```

Comment:
This todo is poorly explained. Especially in relation ship to the JDoc on this method.

Please expand it.